src/main/java/ui/Ui.java

```diff
@@ -0,0 +1,72 @@
+package ui;
+
+import command.Command;
+import mugexception.MugException;
+import parser.Parser;
+import tasks.TaskList;
+
+/**
+ * User interaction.
+ */
+public class Ui {
+
+    /**
+     * Read the input command.
+     *
+     * @param input User input.
+     * @param tasks Task list.
+     * @return Messages according to the user input.
+     */
+    public String readCommand(String input, TaskList tasks) {
+        String output;
+
+        try {
+            String[] splitOrder = input.split(" ", 2);
+            Command command = Parser.command(splitOrder[0]);
+
+            switch (command) {
+            case LIST:
+                output = tasks.readList();
+                break;
+            case BYE:
+                output = "** Bye. Hope to see you soon!! **";
+                break;
+            case DONE:
+                Parser.input(command, splitOrder.length, false);
+                Parser.info(command, splitOrder[1], false);
+                int doneTaskId = Parser.index(splitOrder[1], splitOrder.length);
+                output = tasks.taskDone(doneTaskId);
+                break;
+            case DELETE:
+                Parser.input(command, splitOrder.length, false);
+                Parser.info(command, splitOrder[1], false);
+                int deleteTaskId = Parser.index(splitOrder[1], splitOrder.length);
+                output = tasks.deleteTask(deleteTaskId);
+                break;
+            case TODO:
+            case DEADLINE:
+            case EVENT:
+                Parser.input(command, splitOrder.length, false);
+                Parser.info(command, splitOrder[1], false);
+                String info = splitOrder[1];
+                output = tasks.addTask(command, info);
+                break;
+            case FIND:
+                Parser.input(command, splitOrder.length, false);
+                Parser.info(command, splitOrder[1], false);
+                String keyword = splitOrder[1];
+                output = tasks.searchTask(keyword);
+                break;
```

Comment:
consider splitting these parts out into private methods for readibility purposes

Reply:
Yup, I agree with it too I had split this for week 5 IP. Thank you for the advice.